dtschema/schemas/chosen.yaml

```diff
@@ -206,6 +206,12 @@ properties:
       This property can be used by U-Boot to pass the selected configuration unit
       name of the booted image down to the operating system.
 
+  u-boot,version:
+    $ref: types.yaml#/definitions/string
```

Comment:
How about adding a regex for the version pattern?

Reply:
I think any printable string should be valid, maybe we could exclude spaces.

Just a few examples from some device I have switched on at the moment: `2020.04-5.6.0-devel+git.7d1febd4af77`, `2020.07-00120-g5d6c26ab56bb`, `2022.04`. In addition to that in the past even the minimal/basic U-Boot version had a different structure (`1.2.0`).

Is it worth doing it? What do you think @trini, am I wrong?

Reply:
> I think any printable string should be valid, maybe we could exclude spaces.
> 
> Just a few examples from some device I have switched on at the moment: `2020.04-5.6.0-devel+git.7d1febd4af77`, `2020.07-00120-g5d6c26ab56bb`, `2022.04`. In addition to that in the past even the minimal/basic U-Boot version had a different structure (`1.2.0`).
> 
> Is it worth doing it? What do you think @trini, am I wrong?

To be clear, we always start with YEAR.MONTH but can have a ".SUBLEVEL" instead of -rcN (or empty) and then we have the same CONFIG_LOCALVERSION logic as the linux kernel does, and scripts/setlocalversion copied as well.